src/notebooks/outputs/tracebackFormatter.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { injectable } from 'inversify';
+import { NotebookCell } from 'vscode';
+import { ITracebackFormatter } from '../../kernels/types';
+import { getFilePath } from '../../platform/common/platform/fs-paths';
+import { DataScience } from '../../platform/common/utils/localize';
+import { JupyterNotebookView } from '../constants';
+
+// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+const LineNumberMatchRegex = /(;32m[ ->]*?)(\d+)(.*)/g;
+
+@injectable()
+export class NotebookTracebackFormatter implements ITracebackFormatter {
+    format(cell: NotebookCell, traceback: string[]): string[] {
+        if (cell.notebook.notebookType !== JupyterNotebookView) {
+            return traceback;
+        }
+        return traceback.map((line) => {
+            const inputMatch = /^Input.*?\[.*32mIn\s+\[(\d+).*?0;36m(.*?)\n.*/.exec(line);
```

Comment:
Why would notebook be different than interactive window? Tracebacks should be the same?

Reply:
As in this 'Input' match should be in both. Although I thought it was in the cell hash provider before.

Reply:
Oh yeah it is still there, but it seems like the notebook could be skipping some of the other checks for tracebacks. Like for IPython7 tracebacks. They don't have this form.

Reply:
I was thinking you could refactor this into a single traceback formatter that depends upon something that gives hashes for executed code (ICellHashProvider) and takes an optional notebookCell as an argument.

Reply:
That new tracebackformatter would have all of the 'modifyTraceback' code currently in the cell hash provider.

Reply:
Yes we can do that, however for notebook traceback formatting, 90% of the stuff thats in cell has provider.
Its super simple. I can however try move some of that into a seprate class/function and share that between the two. 

But i don't want to use the same class, else Notebook traceback formatter will end up carrying a lot of baggage which isn't required at all. 

Lets think of it this way, its sepration of concerns.
If lets say IW is never going to be supported in Web (hypothetical scenario), we cannot do this as we have a lot of other dependencies in this class, and we make the optional, etc. lots of work arounds. 

Also if you look at the existing code we have a very clear `if else` condition that explicitly handles notebook tracebacks (two very different code paths), and that works very well to have a seprate class than have it in the same class to handle everything.

Reply:
> Like for IPython7 tracebacks. They don't have this form.

I tested the existing code, the extension doesn't handle IPython7 tracebacks when working in notebooks at all. If you test in stable, we don't do anything at all.
I.e. its already broken. Will file a bug for this.

Though if we wanted to fix this, then we'd need the cell hash generators again.

I guess we need a single traceback formatter in that case. Ok, I'll move this back into a single traceback formatter then.

Reply:
I do think it's good we have a separate class for the traceback formatter (agree with separation of concerns). I've always wanted to refactor this to a separate class. 

Just worried we might be missing something in the super simplified notebook one. 

I think the traceback formatter doesn't need to know about notebooks or kernels though? Just needs the cellhashprovider and a pointer to the actual notebook cell when formatting. Wouldn't that be enough to find the correct hashes? Oh maybe not for pointers to other cells.